device/common/include/traccc/finding/device/propagate_to_next_surface.hpp

```diff
@@ -76,6 +76,11 @@ struct propagate_to_next_surface_payload {
      * input seed
      */
     vecmem::data::vector_view<unsigned int> n_tracks_per_seed_view;
+
+    /**
+     * @brief Optional debug object
+     */
+    propagate_to_next_surface_debug* debug;
```

Comment:
```suggestion
    propagate_to_next_surface_debug* debug = nullptr;
```

Then you wouldn't need to touch the SYCL code if you don't want to.